src/sentry/incidents/grouptype.py

```diff
@@ -27,29 +35,136 @@ class MetricIssueEvidenceData(EvidenceData):
+                evidence_data={
+                    "detector_id": self.detector.id,
+                    "value": priority,
+                    "data_condition_ids": [detector_trigger.id],
+                    "data_condition_type": detector_trigger.type,
+                    "data_condition_comparison_value": self.extract_value(data_packet),
+                    "alert_id": alert_id,
+                },
+                evidence_display=[],  # XXX: may need to pass more info here for the front end
+                type=MetricIssue,
+                level="error",
+                culprit="",
+                priority=priority,
+                assignee=assignee,
+            ),
+            event_data,
         )
-        return occurrence, {}
 
     def extract_dedupe_value(self, data_packet: DataPacket[QuerySubscriptionUpdate]) -> int:
         return int(data_packet.packet.get("timestamp", datetime.now(UTC)).timestamp())
 
     def extract_value(self, data_packet: DataPacket[QuerySubscriptionUpdate]) -> int:
         return data_packet.packet["values"]["value"]
 
+    def construct_title(
+        self,
+        snuba_query: SnubaQuery,
+        detector_trigger: DataCondition,
+        priority: DetectorPriorityLevel,
+    ) -> str:
+        comparison_delta = self.detector.config.get("comparison_delta")
+        agg_display_key = snuba_query.aggregate
+
+        if is_mri_field(agg_display_key):
+            aggregate = format_mri_field(agg_display_key)
+        elif CRASH_RATE_ALERT_AGGREGATE_ALIAS in agg_display_key:
+            agg_display_key = agg_display_key.split(f"AS {CRASH_RATE_ALERT_AGGREGATE_ALIAS}")[
+                0
+            ].strip()
+            aggregate = QUERY_AGGREGATION_DISPLAY.get(agg_display_key, agg_display_key)
+        else:
+            aggregate = QUERY_AGGREGATION_DISPLAY.get(agg_display_key, agg_display_key)
+
+        # Determine the higher or lower comparison
+        higher_or_lower = ""
+        if detector_trigger.type == Condition.GREATER:
+            higher_or_lower = "greater than" if comparison_delta else "above"
+        else:
+            higher_or_lower = "less than" if comparison_delta else "below"
+
+        label = "Warning" if priority == DetectorPriorityLevel.MEDIUM else "Critical"
+
+        # Format the time window for the threshold
+        time_window = format_duration_idiomatic(snuba_query.time_window // 60)
+
+        # If the detector_trigger has a comparison delta, format the comparison string
+        comparison: str | int | float = "threshold"
+        if comparison_delta:
+            comparison_delta_minutes = comparison_delta // 60
+            comparison = TEXT_COMPARISON_DELTA.get(
+                comparison_delta_minutes, f"same time {comparison_delta_minutes} minutes ago "
+            )
+        else:
+            comparison = detector_trigger.comparison
+
+        template = "{label}: {metric} in the last {time_window} {higher_or_lower} {comparison}"
```

Comment:
just want to confirm that there won't be extra spaces at the end if we don't have a comparison value? (i think that's the only one that is potentially not set)

do we need to worry about internationalization on this string as well?

Reply:
There are tests that show the title with and without the comparison delta and I didn't see extra spaces. I'll try to find what we use for translation.

Reply:
I can't find any instance of us translating strings in the backend code, I know we use transifex to do translations for stuff in the app but I can't find evidence of messaging stuff being translated - we should do it but I'm not too worried about it for this one string.